### PR TITLE
Updated localisation system

### PR DESCRIPTION
**Pull request by sammacbeth**
Updates tooling around translations to make it easier to isolate localized strings and push them to a translation service.
 
For each locale we now have a `l10n/{locale}.json` file which defines strings which we have overridden from Firefox defaults. These are then used by fern to patch `mozilla-release` and `l10n/{locale}/*`. `l10n/en-US.json` acts as a master file from which other translations can be generated by the translation service.

As the en-US strings are now all commited in a fern managed patch, we have to remove all string changes from manual patches. This allowed several patches to be dropped entirely.

We retain the ability to do manual patches for specific locales, which we can use, for example for default bookmarks.

### CODE REVIEW

File: fern/patches/l10n.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        lines[i] = `${key}${separator}${replacements[key].string}`;
+      }
+    }
+  }
+  return lines.join("\n");
+}
+
+module.exports = ({ locale = "en-US" }) => ({
+  name: "Rewrite localized strings",
+  paths: ["./*"],
```
Comment (chrmod) [inline fern/patches/l10n.js:52]: is this needed? shouldn't `skip: () => false` do the job?

File: patches/.index [hunk 2]
```diff
@@ -19,23 +19,19 @@
...
+0028-Allow-update-channel-to-be-configured-via-prefs.patch
+0029-Add-DNS-and-websocket-permissions-to-ghostery-extens.patch
+0030-Fix-compiler-error-on-optimised-windows-builds.patch
+0031-Feather-icons.patch
+0032-Update-text-on-about-license.patch
+0033-Hide-addon-recommendation-badge.patch
+0034-Disable-ion-pioneer-studies.patch
+0035-Hide-learn-more-link-on-network-error-pages.patch
+0036-Remove-Firefox-strings-for-new-install-page.patch
+0037-OpenURL-on-buildID-changes.patch
```
Comment (chrmod) [inline patches/.index:37]: Sweet!

**Review by chrmod — APPROVED** [review]
Looks great!

File: fern/patches/l10n.js (hunk 1 above)
Comment (sammacbeth) [reply fern/patches/l10n.js:52]: It is. The `git add` later explicitly uses the paths from this property.

**Pull request merged**